custom-targets/vertex-ai/model-deployer/request.go

```diff
@@ -0,0 +1,198 @@
+	switch reqType {
+	case "RENDER":
+		return newRenderHandler(gcsClient)
+	case "DEPLOY":
+		return newDeployHandler(gcsClient)
+	default:
+		return nil, fmt.Errorf("received unexpected Cloud Deploy params type: %v", reqType)
+	}
+}
+
+// newRenderHandler returns a handler for processing render requests.
+func newRenderHandler(gcsClient *storage.Client) (requestHandler, error) {
+	percentage, err := strconv.Atoi(os.Getenv(clouddeploy.PercentageEnvKey))
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse env var %q", clouddeploy.PercentageEnvKey)
+	}
+
+	params, err := determineParams()
+
+	if err != nil {
+		return nil, fmt.Errorf("unable to parse params: %v", err)
+	}
+
+	rr := &clouddeploy.RenderRequest{
+		Target:        os.Getenv(clouddeploy.TargetEnvKey),
+		Percentage:    percentage,
+		InputGCSPath:  os.Getenv(clouddeploy.InputGCSEnvKey),
+		OutputGCSPath: os.Getenv(clouddeploy.OutputGCSEnvKey),
+	}
+
+	return &renderer{
+		gcsClient: gcsClient,
+		req:       rr,
+		params:    params}, nil
+}
+
+// newDeployHandler returns a handler for processing deploy requests.
+func newDeployHandler(gcsClient *storage.Client) (requestHandler, error) {
+	percentage, err := strconv.Atoi(os.Getenv(clouddeploy.PercentageEnvKey))
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse env var %q: %v", clouddeploy.PercentageEnvKey, err)
+	}
+
+	params, err := determineParams()
+
+	if err != nil {
+		return nil, fmt.Errorf("unable to parse params: %v", err)
+	}
+
+	request := &clouddeploy.DeployRequest{
+		Percentage:      percentage,
+		ManifestGCSPath: os.Getenv(clouddeploy.ManifestGCSEnvKey),
+		OutputGCSPath:   os.Getenv(clouddeploy.OutputGCSEnvKey),
+	}
+	return &deployer{gcsClient: gcsClient, req: request, params: params}, nil
+}
+
+// params contains the deploy parameter values passed into the execution environment.
+type params struct {
+	// The minimum replica count for the deployed model obtained via a deploy parameter
+	minReplicaCount int64
+
+	// The model to be deployed. May or may not contain a tag or version number.
+	// format is "projects/{project}/locations/{location}/models/{modelId}[@versionId|alias].
+	model string
+
+	// The endpoint where the model will be deployed
+	// format is "projects/{project}/locations/{location}/endpoints/{endpointId}.
+	endpoint string
+
+	// directory path where the renderer should look for target-specific configuration
+	// for this deployment, if not provided the renderer will check for a deployModel.yaml
+	// fie in the root working directory.
+	configPath string
+}
+
+// determineParams returns the supported params provided in the execution environment via environment variables.
+func determineParams() (*params, error) {
+
+	replicaCount, err := strconv.Atoi(os.Getenv(minReplicaCountEnvKey))
+	if err != nil {
+		fmt.Println("unable to obtain minReplicaCount parameter from environment variable, will attempt to check if in the configuration file if its provided")
```

Comment:
I would remove this log and when validating at render just log that it wasn't provided as deploy parameter or in the configuration file.